src/client/app/+home/home.component.ts

```diff
@@ -13,45 +17,49 @@ import { GraphPreviewListService } from '../shared/index';
 
 export class HomeComponent implements OnInit {
 
-  newGraphPreview: string = '';
-  errorMessage: string;
-  graphPreviews: any[] = [];
+  userLoggedIn: boolean = false;
+  graphPreviews: DriveFile[] = [];
 
   /**
    * Creates an instance of the HomeComponent with the injected
    * GraphPreviewListService.
    *
-   * @param {GraphPreviewListService} GraphPreviewListService - The injected GraphPreviewListService.
+   * @param googleRealtimeService
+   * @param changeDetector
    */
-  constructor(public graphPreviewListService: GraphPreviewListService) {}
+  constructor(private googleRealtimeService: GoogleRealtimeService,
+              private changeDetector: ChangeDetectorRef,
+              private router: Router) {
+  }
 
   /**
    * Get the names OnInit
    */
   ngOnInit() {
-    this.getGraphPreviews();
+    this.googleRealtimeService.oauthToken.subscribe((oauthToken) => {
```

Comment:
Yup, I never cleaned up after myself in my component code (not necessary for Service code since they're singletons that live forever anyways).

Done.